### PR TITLE
Populate IndexingTask table after creation (#5466)

### LINKED ISSUE #5466: Lucene indexation : content items not indexed if created before index is created
When we create a Lucene index after having creating some content items it won't index them because they did not get added to the IndexingTask table which gets added to the database only if you enable the Lucene module. So to repro. 

Create some content items.
Enable the Lucene feature.
Create a Lucene Index.
Hit rebuild index.

The content items won't be indexed. So we should add a new button on the Index List to resync the content items that should be added to the IndexingTask table. This button could also flush the ContentItems that should not be there at the same time.

### PR DESCRIPTION
**Pull request by AndreySurkov**
Proposed solution to fix #5466. When OrchardCore.Indexing module enabled, IndexingTask table should contain record for each ContentItem. But if I enable OrchardCore.Indexing module when system already has content items, i need to republish them to make them available for indexing. That is not so simple if you have hundreds of items.  This change will populate  table with required data during creation.  Or in case if system already has gaps in IndexingTask table it will be fixed during migration from version 1 .

### CODE REVIEW

**Comment by Piedone** [comment]
@Skrypt is this something you could review?

**Comment by Skrypt** [comment]
I would rather have a button in the admin UI to execute that same SQL Query. Else, we never know when someone will disable the OrchardCore.Indexing feature and enable it back a couple days later. Then you will miss all the documents published in these days and will have no way to execute this Query again. Also, it should insert only records that are not already there to not create duplicates.

**Comment by AndreySurkov** [comment]
@Skrypt  It is a good point to have ability to run it in any time. Do you have any suggestion where this button should be placed?

As for "_Also, it should insert only records that are not already there to not create duplicates._" - it is already covered in this PR.

**Comment by Skrypt** [comment]
I think you will need to create a new configuration page in the admin UI.

Configurations --> Indexing

New drivers and views in the OrchardCore.Indexing module.

File: src/OrchardCore.Modules/OrchardCore.Indexing/Migrations.cs [hunk 1]
```diff
@@ -19,6 +27,69 @@ await SchemaBuilder.AlterTableAsync(nameof(IndexingTask), table => table
...
+
+                var indexingSelectorBuildr = dialect.CreateBuilder(tablePrefix);
+                indexingSelectorBuildr.Select();
+                indexingSelectorBuildr.Selector(nameof(IndexingTask), nameof(IndexingTask.ContentItemId), schema);
+                indexingSelectorBuildr.Table(nameof(IndexingTask), null, schema);
+
+                var sqlBuilder = dialect.CreateBuilder(tablePrefix);
+                sqlBuilder.Select();
+                sqlBuilder.Selector(nameof(ContentItemIndex), nameof(ContentItemIndex.ContentItemId), schema);
+                sqlBuilder.AddSelector(", @CreatedUtc, 0");
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Indexing/Migrations.cs:76]: Are you sure this `@` will work in all databases? Isn't there a helper in SqlBuilder? Or did you find the same thing somewhere else?

**Comment by sebastienros** [comment]
> Do you have any suggestion where this button should be placed?

This can be kept automatic if you implement a different event than migrations. There is an event to detect a feature is enabled (this is what migrations use).

**Pull request closed without merging**